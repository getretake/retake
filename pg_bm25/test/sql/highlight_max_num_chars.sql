-- Default highlighting without max_num_chars
SELECT description, rating, category, paradedb.rank_bm25(ctid), paradedb.highlight_bm25(ctid, 'description')
FROM paradedb.mock_items
WHERE mock_items @@@ 'description:keyboard OR category:electronics'
LIMIT 5;

-- max_num_chars is set to 14 
SELECT description, rating, category, paradedb.rank_bm25(ctid), paradedb.highlight_bm25(ctid, 'description')
FROM paradedb.mock_items
WHERE mock_items @@@ 'description:keyboard OR category:electronics:::max_num_chars=14'
LIMIT 5;
