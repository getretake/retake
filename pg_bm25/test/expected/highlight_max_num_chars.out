-- Default highlighting without max_num_chars
SELECT description, rating, category, paradedb.rank_bm25(ctid), paradedb.highlight_bm25(ctid, 'description')
FROM paradedb.mock_items
WHERE mock_items @@@ 'description:keyboard OR category:electronics'
LIMIT 5;
         description         | rating |  category   | rank_bm25 |         highlight_bm25          
-----------------------------+--------+-------------+-----------+---------------------------------
 Ergonomic metal keyboard    |      4 | Electronics |   4.93101 | Ergonomic metal <b>keyboard</b>
 Plastic Keyboard            |      4 | Electronics |    5.3765 | Plastic <b>Keyboard</b>
 Innovative wireless earbuds |      5 | Electronics |   2.10964 | 
 Fast charging power bank    |      4 | Electronics |   2.10964 | 
 Bluetooth-enabled speaker   |      3 | Electronics |   2.10964 | 
(5 rows)

-- max_num_chars is set to 14 
SELECT description, rating, category, paradedb.rank_bm25(ctid), paradedb.highlight_bm25(ctid, 'description')
FROM paradedb.mock_items
WHERE mock_items @@@ 'description:keyboard OR category:electronics:::max_num_chars=14'
LIMIT 5;
         description         | rating |  category   | rank_bm25 |    highlight_bm25     
-----------------------------+--------+-------------+-----------+-----------------------
 Ergonomic metal keyboard    |      4 | Electronics |   4.93101 | metal <b>keyboard</b>
 Plastic Keyboard            |      4 | Electronics |    5.3765 | <b>Keyboard</b>
 Innovative wireless earbuds |      5 | Electronics |   2.10964 | 
 Fast charging power bank    |      4 | Electronics |   2.10964 | 
 Bluetooth-enabled speaker   |      3 | Electronics |   2.10964 | 
(5 rows)

